packages/eas-cli/src/credentials/ios/appstore/bundleIdCapabilities.ts

```diff
@@ -0,0 +1,415 @@
+import {
+  BundleId,
+  CapabilityOptionMap,
+  CapabilityType,
+  CapabilityTypeDataProtectionOption,
+  CapabilityTypeOption,
+} from '@expo/apple-utils';
+import { JSONObject, JSONValue } from '@expo/json-file';
+
+import Log from '../../../log';
+
+type GetOptionsMethod<T extends CapabilityType = any> = (
+  entitlement: JSONValue,
+  entitlementsJson: JSONObject
+) => CapabilityOptionMap[T];
+
+const validateBooleanOptions = (options: any): boolean => {
+  return typeof options === 'boolean';
+};
+
+const validateStringArrayOptions = (options: any): boolean => {
+  return Array.isArray(options) && options.every(option => typeof option === 'string');
+};
+
+const createValidateStringOptions = (allowed: string[]) => (options: any): boolean => {
+  return allowed.includes(options);
+};
+
+const createValidateStringArrayOptions = (allowed: string[]) => (options: any): boolean => {
+  return Array.isArray(options) && options.every(option => allowed.includes(option));
+};
+
+const validateDevProdString = createValidateStringOptions(['development', 'production']);
+
+const getBooleanOptions: GetOptionsMethod = entitlement => {
+  return entitlement === true ? CapabilityTypeOption.ON : CapabilityTypeOption.OFF;
+};
+
+const getDefinedOptions: GetOptionsMethod = entitlement => {
+  return entitlement ? CapabilityTypeOption.ON : CapabilityTypeOption.OFF;
+};
+
+export async function syncCapabilitiesForEntitlementsAsync(
+  bundleId: BundleId,
+  entitlements: JSONObject
+) {
+  const capabilities = await bundleId.getBundleIdCapabilitiesAsync();
+  const enabled: string[] = [];
+  const disabled: string[] = [];
+  const request: { capabilityType: CapabilityType; option: any }[] = [];
+  for (const [key, value] of Object.entries(entitlements)) {
+    if (value) {
+      const capability = CapabilityMapping.find(capability => capability.entitlement === key);
```

Comment:
```suggestion
      const capabilityDefinition = CapabilityMapping.find(capability => capability.entitlement === key);
```
or similar

It made me confused, because when I was reading the later part of this code I thought it's a member of `capabilities`.